traveler-ios-ui/TravelerKitUI/ViewControllers/BookingItemSearch/BookingItemSearchSortFilterViewController.swift

```diff
@@ -0,0 +1,189 @@
+        
+        tableView.register(UINib(nibName: "BookingItemSearchSortFilterCell", bundle: Bundle(for: BookingItemSearchSortFilterViewController.self)), forCellReuseIdentifier: CellIdentifiers.bookingItemSearchSortFilter)
+        
+        resetData()
+    }
+    
+    open override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        switch (segue.identifier, segue.destination) {
+        case ("sortSegue", let navVC as UINavigationController):
+            let vc = navVC.topViewController as? BookingItemSearchSortViewController
+            vc?.delegate = self
+            vc?.currentSelection = currentSort!
+        case ("filterPriceSegue", let navVC as UINavigationController):
+            let vc = navVC.topViewController as? BookingItemSearchFilterPriceViewController
+            vc?.delegate = self
+            vc?.currentRange = currentRange
+            vc?.fullRange = fullRange
+        default:
+            Log("Unknown segue", data: segue, level: .warning)
+            break
+        }
+    }
+    
+    @IBAction func cancel() {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @IBAction func submit() {
+        guard let currentSearchFacets = currentSearchFacets, searchQuery != nil else {
+            cancel()
+            return
+        }
+        
+        let sortData = currentSort!.data
+        var bookingItemSearchFilters = BookingItemSearchFilters()
+        bookingItemSearchFilters.priceRange = PriceRangeFilter(range: currentRange!, currency: currentSearchFacets.minPrice.baseCurrency)
+        bookingItemSearchFilters.categories = searchQuery!.categories
+        bookingItemSearchFilters.city = searchQuery!.city
+        bookingItemSearchFilters.country = searchQuery!.country
+        bookingItemSearchFilters.sortOption = sortData.sortOption
+        bookingItemSearchFilters.sortOrder = sortData.sortOrder
+        searchQuery = searchQuery!.filterSearchWith(bookingItemSearchFilters)
+        delegate?.bookingItemSearchSortFilterController(self, didSubmit: searchQuery!)
+    }
+    
+    func resetData() {
+        currentSort = defaultSort
+        
+        if let minFacetsPrice = currentSearchFacets?.minPrice.valueInBaseCurrency, let maxFacetsPrice = currentSearchFacets?.maxPrice.valueInBaseCurrency, minFacetsPrice <= maxFacetsPrice {
+            currentRange = minFacetsPrice...maxFacetsPrice
+        } else {
+            currentRange = fullRange
+        }
+    }
+    
+    @IBAction func reset() {
+        resetData()
+        tableView.reloadData()
+    }
+    
+    public override func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+    
+    public override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch section {
+        case 0:
+            return 1
+        case 1:
+            return 1
+        default:
+            fatalError("Unhandled section; this should not happen!")
```

Comment:
Its bad practice to crash the app if an unhandled section occurs. This will crash the airline's app and it will look bad on us. Use log instead and return nil or 0 or something else.